actors/systemsettings/src/main/java/org/sunbird/systemsettings/actors/SystemSettingsActor.java

```diff
@@ -86,8 +85,67 @@ private void setSystemSetting(Request actorMessage) {
     ProjectLogger.log("SystemSettingsActor: setSystemSetting called", LoggerEnum.DEBUG.name());
 
     Map<String, Object> req = actorMessage.getRequest();
+    if (((String) req.get(JsonKey.FIELD)).equalsIgnoreCase(JsonKey.PHONE_UNIQUE)) {
+      checkEmailOrPhoneData(
+          (String) req.get(JsonKey.FIELD),
+          (String) req.get(JsonKey.VALUE),
+          JsonKey.ENC_PHONE,
+          ResponseCode.duplicatePhoneData,
+          JsonKey.PHONE);
+    }
+    if (((String) req.get(JsonKey.FIELD)).equalsIgnoreCase(JsonKey.EMAIL_UNIQUE)) {
+      checkEmailOrPhoneData(
+          (String) req.get(JsonKey.FIELD),
+          (String) req.get(JsonKey.VALUE),
+          JsonKey.ENC_EMAIL,
+          ResponseCode.duplicateEmailData,
+          JsonKey.EMAIL);
+    }
     SystemSetting systemSetting = mapper.convertValue(req, SystemSetting.class);
     Response response = systemSettingDaoImpl.write(systemSetting);
     sender().tell(response, self());
   }
+
+  private void checkEmailOrPhoneData(
```

Comment:
Move this code to UserClient with below public methods.

(public methods)
esIsEmailUnique
esIsPhoneUnique

(private method)
esIsFieldUnique